### PR TITLE
Update test setup Windows compatible

### PR DESCRIPTION
**Pull request by kwonoj**
This PR adds small updates in test fixture / runner setups, make development pipeline scripts are working on Windows OS environment as well. Non-windows regression verification was done via travis at https://travis-ci.org/kwonoj/react/jobs/187317960 .

- relates to #7849

### CODE REVIEW

File: src/isomorphic/classic/types/__tests__/ReactPropTypesProduction-test.js [hunk 1]
```diff
@@ -23,7 +23,8 @@ describe('ReactPropTypesProduction', function() {
     __DEV__ = false;
     oldProcess = process;
     global.process = {
-      env: Object.assign({}, process.env, {NODE_ENV: 'production'}),
+      cwd: process.cwd,
```
Comment (kwonoj) [inline src/isomorphic/classic/types/__tests__/ReactPropTypesProduction-test.js]: Don't know reason why, but on Windows when `global.process` has replaced it loses access to `process.cwd`, test runner setup fails to use those function.

**Comment by gaearon** [comment]
Hey, sorry this slipped through the cracks. Is this still an issue?

**Comment by kwonoj** [comment]
@gaearon so far I believe it's still there. I'll try to rebase today.

**Comment by gaearon** [comment]
Thanks!

**Comment by kwonoj** [comment]
rebased to latest master.

File: src/isomorphic/classic/types/__tests__/ReactPropTypesProduction-test.js [hunk 2]
```diff
@@ -23,7 +23,8 @@ describe('ReactPropTypesProduction', function() {
...
+      env: Object.assign({}, process.env, {NODE_ENV: JSON.stringify('production')}),
```
Comment (gaearon) [inline src/isomorphic/classic/types/__tests__/ReactPropTypesProduction-test.js]: I don't think you need `JSON.stringify` here. This is not Webpack environment, it's setting a specific value. So you don't want to add extra quotes around it.

Comment (kwonoj) [inline src/isomorphic/classic/types/__tests__/ReactPropTypesProduction-test.js]: yeah, good catch. I think I saw some glitch when not to stringify, but it seems not - updated.

**Comment by gaearon** [comment]
Thanks!

**Pull request merged**